From df4c3642b5c758fb5c934463fe8425be53328e4b Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Tue, 7 Oct 2025 19:37:57 +0300
Subject: [PATCH 358/553] tst_QDir: port away from QPair

Pick-to: 6.8 6.5
Task-number: QTBUG-115841
Change-Id: I88189a0f41470b39695b41c12478feb9f4dd971c
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit fd25c0c81a358cc47807453c7825dec460089026)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 tests/auto/corelib/io/qdir/tst_qdir.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/auto/corelib/io/qdir/tst_qdir.cpp b/tests/auto/corelib/io/qdir/tst_qdir.cpp
index 64be9797845..2b051e9a9cb 100644
--- a/tests/auto/corelib/io/qdir/tst_qdir.cpp
+++ b/tests/auto/corelib/io/qdir/tst_qdir.cpp
@@ -987,14 +987,14 @@ void tst_QDir::entryListWithTestFiles()
 #if defined(Q_OS_WIN)
     // ### Sadly, this is a platform difference right now.
     // Note we are using capital L in entryList on one side here, to test case-insensitivity
-    const QList<QPair<QString, QString> > symLinks =
+    const QList<std::pair<QString, QString>> symLinks =
     {
         {m_dataPath + "/entryList/file", entrylistPath + "linktofile.lnk"},
         {m_dataPath + "/entryList/directory", entrylistPath + "linktodirectory.lnk"},
         {m_dataPath + "/entryList/nothing", entrylistPath + "brokenlink.lnk"}
     };
 #else
-    const QList<QPair<QString, QString> > symLinks =
+    const QList<std::pair<QString, QString>> symLinks =
     {
         {"file", entrylistPath + "linktofile.lnk"},
         {"directory", entrylistPath + "linktodirectory.lnk"},
-- 
2.51.2

